pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -368,22 +359,12 @@ private CompletableFuture<Void> revokePermissionsAsync(String topicUri, String r
     protected void internalRevokePermissionsOnTopic(AsyncResponse asyncResponse, String role) {
         // This operation should be reading from zookeeper and it should be allowed without having admin privileges
         validateAdminAccessForTenantAsync(namespaceName.getTenant())
-                .thenCompose(__ -> validatePoliciesReadOnlyAccessAsync().thenCompose(unused1 ->
-            getPartitionedTopicMetadataAsync(topicName, true, false)
-                .thenCompose(metadata -> {
-                    int numPartitions = metadata.partitions;
+                .thenCompose(__ -> validatePoliciesReadOnlyAccessAsync().thenCompose(unused1 -> {
                     CompletableFuture<Void> future = CompletableFuture.completedFuture(null);
-                    if (numPartitions > 0) {
-                        for (int i = 0; i < numPartitions; i++) {
-                            TopicName topicNamePartition = topicName.getPartition(i);
-                            future = future.thenComposeAsync(unused ->
-                                    revokePermissionsAsync(topicNamePartition.toString(), role));
-                        }
-                    }
-                    return future.thenComposeAsync(unused -> revokePermissionsAsync(topicName.toString(), role))
+                    return future.thenComposeAsync(unused ->
```

Comment:
```suggestion
                    return revokePermissionsAsync(topicName.getPartitionedTopicName(), role)
```

Reply:
You also need to check whether the topic is partitioned topic, then remove the permissions of each partition topic.